server/src/utils/typedObject.ts

```diff
@@ -0,0 +1,3 @@
+export function fromEntries<T>(entries: [keyof T, T[keyof T]][]): T {
+  return entries.reduce((acc, [key, value]) => ({ ...acc, [key]: value }), <T>{});
+}
```

Comment:
Did Object.fromEntries<T> not work? I don't believe you need to write this.

Reply:
No it didn't really work. Object.fromEntries<T> casts the value to whatever key to T.

Reply:
I see, then you do need to cast the return of that native function then, I misread the type signature.